### ISSUE TITLE
Parameter should use code inline block

### ISSUE THREAD

**Issue by X-Ryl669** [matusnovak/doxybook#9]
Current, if you document a function like this:
```
/** Some brief here.
     @param a   Something 
     @param foo The bar stuff
     @return true if moon shines */
bool myFunc(int a, char foo);
```
You'll get a very ugly result in markdown:

* a Something
* foo The bar stuff

I think it'd be much better if the parameter name would stand up, that is, like this:

* `a` Something
* `foo` The bar stuff

Or even better, if they were aligned (easier to read), like this:

| Parameter | Description |
|---|---|
| `a` | Something |
| `foo` | The bar stuff |

If the header is *too much*, then, I guess this is also acceptable:

| | |
|---|---|
| `a` | Something |
| `foo` | The bar stuff |

Same issue with @return presentation, it's too fade.

**Comment by matusnovak** [comment]
Hi @X-Ryl669 

I don't think a table would be good. I have tried that before and the output became hard to navigate, specially if you have many parameters. I really do not like that. 

But I like the idea of using `monospaced` for the `@params`. The changes are in [improve-parameterlist-readability](https://github.com/matusnovak/doxybook/tree/improve-parameterlist-readability) branch. This also changes the template parameters and return value lists. I have added some screenshots below. Does this look better?

**MkDocs:**

![2019-11-12 19_38_29-Class example__Animal - My Docs](https://user-images.githubusercontent.com/10331278/68700459-3c60fd80-0585-11ea-938a-8f0184fb342b.png)
![2019-11-12 19_53_52-Class example__Animal - My Docs](https://user-images.githubusercontent.com/10331278/68700983-2e5fac80-0586-11ea-8aa9-480333ad32da.png)

**GitBook:**

![2019-11-12 19_39_25-class example__Animal · GitBook](https://user-images.githubusercontent.com/10331278/68700461-3e2ac100-0585-11ea-93b9-cde91e150ae4.png)

**Vuepress:**

![2019-11-12 19_41_35-Class example__Animal _ Doxybook Example Output](https://user-images.githubusercontent.com/10331278/68700465-3f5bee00-0585-11ea-9e72-edff39cf3fbf.png)

**Comment by X-Ryl669** [comment]
I've tried it and it's great. I could not get return value to work however like in your example. Maybe I've something wrong in my setup. Thanks!

**Comment by matusnovak** [comment]
@X-Ryl669  That's strange. Would you be able to send me XML file of that particular piece? 

I need something like this (was used to create one of the screenshot above):

```xml
<detaileddescription>
    <para>
        <simplesect kind="see">
            <para>
                <ref refid="classexample_1_1_animal_interface_1adf3678fbed03da52a7bcbd3cd6a41d57" kindref="member">get_num_of_limbs</ref>
            </para>
        </simplesect>
        <parameterlist kind="retval">
            <parameteritem>
                <parameternamelist>
                    <parametername>true</parametername>
                </parameternamelist>
                <parameterdescription>
                    <para>Does have a tail </para>
                </parameterdescription>
            </parameteritem>
            <parameteritem>
                <parameternamelist>
                    <parametername>false</parametername>
                </parameternamelist>
                <parameterdescription>
                    <para>Does not have a tail </para>
                </parameterdescription>
            </parameteritem>
        </parameterlist>
    </para>
</detaileddescription>
```

**Comment by X-Ryl669** [comment]
Got this:
```
      <memberdef kind="function" id="struct_n_f_c_1a5c7f8de0ed60384631f084882e99d875" prot="public" static="no" const="no" explicit="no" inline="no" virt="pure-virtual">
        <type>bool</type>
        <definition>virtual bool NFC::waitForMifareTag</definition>
        <argsstring>(Mifare &amp;tag, unsigned timeoutMs)=0</argsstring>
        <name>waitForMifareTag</name>
        <reimplementedby refid="struct_p_n532_1a195494d39979964c3e59fc650e6a3ccf">waitForMifareTag</reimplementedby>
        <param>
          <type><ref refid="struct_mifare" kindref="compound">Mifare</ref> &amp;</type>
          <declname>tag</declname>
        </param>
        <param>
          <type>unsigned</type>
          <declname>timeoutMs</declname>
        </param>
        <briefdescription>
<para>Wait for a <ref refid="struct_mifare" kindref="compound">Mifare</ref> tag and erase the given card with the new card found (if found). </para>        </briefdescription>
        <detaileddescription>
<para>This is a wrapper over readPassiveTargetID. <parameterlist kind="param"><parameteritem>
<parameternamelist>
<parametername>tag</parametername>
</parameternamelist>
<parameterdescription>
<para>If successful, will contains the detected tag. </para></parameterdescription>
</parameteritem>
<parameteritem>
<parameternamelist>
<parametername>timeoutMs</parametername>
</parameternamelist>
<parameterdescription>
<para>The number of millisecond to wait to find a card type. </para></parameterdescription>
</parameteritem>
</parameterlist>
<simplesect kind="return"><para>true if a card was detected and read correctly, false upon error or timeout </para></simplesect>
</para>        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="main/NFC.hpp" line="336" column="1"/>
      </memberdef>
```

Seems like there is no node on the return value here for me (only the description in a `<para>` node) so there's no magic doxybook could do. Mine is `<simplesect kind="return">` while yours is `<parameterlist kind="retval">`. I guess I'll have to look at doxygen configuration here why it's not generating like yours.

Can you confirm you're using `@return true Does have a tail` `@return false Does not have a tail` in your source code ?

**Comment by matusnovak** [comment]
@X-Ryl669 Ahhh, that makes sense. I am using `@retval` instead of `@return`. Like this:

```cpp
/*!
* @brief Returns true if the animal has a tail
* @see get_num_of_limbs
* @retval true Does have a tail
* @retval false Does not have a tail
*/
```

I think it depends based on what are you trying to do. From Doxygen documentation <http://www.doxygen.nl/manual/commands.html#cmdretval> it says:

> Starts a description for a function's return value with name <return value>, followed by a description of the return value. The text of the paragraph that forms the description has no special internal structure. All visual enhancement commands may be used inside the paragraph. Multiple adjacent \retval commands will be joined into a single paragraph. Each return value description will start on a new line. The \retval description ends when a blank line or some other sectioning command is encountered

So I think you should use `@retval` if you want to explicitly say what `true` and `false` does. In that case it should be formatted correctly in Markdown.

**Comment by X-Ryl669** [comment]
Ok, that's clear. I'm closing the issue then.

**Comment by matusnovak** [comment]
The changes have been merged and released as v3.1.4 -> https://pypi.org/project/doxybook/3.1.4/
